src/chronify/store.py

```diff
@@ -132,16 +132,28 @@ def ingest_from_csv(
                 msg = f"IndexTimeRange cannot be converted to {cls_name}"
                 raise NotImplementedError(msg)
 
-        if self.has_table(dst_schema.name):
+        table_exists = self.has_table(dst_schema.name)
+        if table_exists:
             table = Table(dst_schema.name, self._metadata)
         else:
             dtypes = [get_sqlalchemy_type_from_duckdb(x) for x in rel.dtypes]
-            columns = [Column(x, y) for x, y in zip(rel.columns, dtypes)]
-            table = Table(dst_schema.name, self._metadata, *columns)
+            table = Table(
+                dst_schema.name,
+                self._metadata,
+                *[Column(x, y) for x, y in zip(rel.columns, dtypes)],
+            )
             table.create(self._engine)
 
         with self._engine.begin() as conn:
             write_database(rel.to_df(), conn, dst_schema)
+            try:
+                check_timestamps(conn, table, dst_schema)
```

Comment:
Previous behavior was incorrect. We need to perform checks on the final table and rollback on error.